data/org.eclipse.birt.report.data.oda.jdbc.ui/src/org/eclipse/birt/report/data/oda/jdbc/ui/editors/SQLDataSetEditorPage.java

```diff
@@ -1075,4 +1119,149 @@ private void insertTreeItemText() {
+			}
+
+		});
+
+	}
+
+	private void setupFindQueryTextOptions(Group group) {
+
+		GridData csLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		this.findQueryTextCaseSensitive = new Button(group, SWT.CHECK);
+		this.findQueryTextCaseSensitive
+				.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive.tooltip")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive.setSelection(false);
+		this.findQueryTextCaseSensitive.setLayoutData(csLayoutData);
+		this.findQueryTextCaseSensitive.setEnabled(true);
+
+		GridData wwLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		this.findQueryTextWholeWord = new Button(group, SWT.CHECK);
+		this.findQueryTextWholeWord.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word")); //$NON-NLS-1$
+		this.findQueryTextWholeWord
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word.tooltip")); //$NON-NLS-1$
+		this.findQueryTextWholeWord.setSelection(false);
+		this.findQueryTextWholeWord.setLayoutData(wwLayoutData);
+		this.findQueryTextWholeWord.setEnabled(true);
+	}
+
+	private void setupFindQueryTextButtons(Group group) {
+
+		GridData fwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button fwButtonFindText = new Button(group, SWT.BUTTON1);
+		fwButtonFindText.setText("▼ " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward")); //$NON-NLS-1$
+		fwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward.tooltip")); //$NON-NLS-1$
+		fwButtonFindText.setLayoutData(fwButtonFindTextLayoutData);
+		fwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		fwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
+
+					@Override
+					public void run() {
+						findQueryTextForward();
+					}
+				});
+			}
+		});
+
+		GridData bwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button bwButtonFindText = new Button(group, SWT.BUTTON1);
+		bwButtonFindText.setText("▲ " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward")); //$NON-NLS-1$
+		bwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward.tooltip")); //$NON-NLS-1$
+		bwButtonFindText.setLayoutData(bwButtonFindTextLayoutData);
+		bwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		bwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
```

Comment:
The are plenty of control from which to get a display the you don’t need to go to the workbench.

Reply:
Changed, event is simplified without workbench.